### PR TITLE
frontend/#1093/ride reallocation changes to use same estimate id

### PR DESCRIPTION
**Pull request by ashishsingh101**
## Type of Change

- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement
- [x] Refactoring
- [ ] Dependency updates

### Additional Changes

- [ ] This PR modifies the database schema (database migration added)
- [ ] This PR modifies dhall configs/environment variables
- [ ] This PR contains API breaking changes

## Checklist

- [ ] I formatted the code and addressed linter errors `./dev/format-all-files.sh`
- [ ] I reviewed submitted code
- [ ] I added unit tests for my changes where possible
- [ ] I added a [CHANGELOG](/CHANGELOG.md) entry if applicable

### CODE REVIEW

File: Frontend/ui-customer/src/Flow.purs [hunk 1]
```diff
@@ -542,6 +542,30 @@ homeScreenFlow = do
       modifyScreenState $ HomeScreenStateType (\homeScreen -> homeScreen{props{searchId = rideSearchRes.searchId,currentStage = FindingEstimate, rideRequestFlow = true, isSearchLocation = SearchLocation, sourcePlaceId = Nothing, destinationPlaceId = Nothing}})
       updateLocalStage FindingEstimate
       homeScreenFlow
+    RETRY_FINDING_QUOTES -> do
+      void $ lift $ lift $ loaderText (getString LOADING) (getString PLEASE_WAIT_WHILE_IN_PROGRESS)  -- TODO : Handlde Loader in IOS Side
+      void $ lift $ lift $ toggleLoader true
+      (GlobalState newState) <- getState
+      let state = newState.homeScreen
+      when (not (isLocalStageOn QuoteList)) do
+        cancelEstimate state.props.estimateId
```
Comment (ashishsingh101) [inline Frontend/ui-customer/src/Flow.purs:551]: @hkmangla do we need to cancel estimate in any case ?

**Review by witcher-shailesh — APPROVED** [review]

File: Frontend/ui-customer/src/Flow.purs [hunk 2]
```diff
@@ -542,6 +542,34 @@ homeScreenFlow = do
...
+      void $ lift $ lift $ toggleLoader true
+      (GlobalState newState) <- getState
+      let state = newState.homeScreen
+      if (not (isLocalStageOn QuoteList)) then do
+        _ <- pure $ firebaseLogEvent "ny_user_cancel_and_retry_request_quotes"
+        cancelEstimate state.props.estimateId
+      else do
+        _ <- pure $ firebaseLogEvent "ny_user_retry_request_quotes"
+        pure unit
+      _ <- pure $ setValueToLocalStore AUTO_SELECTING "false"
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Flow.purs:556]: why does this function has a bind operation required but same func in next line don't?

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 3]
```diff
@@ -705,6 +705,7 @@ quoteListModelViewState state = { source: state.data.source
                             , selectedQuote: state.props.selectedQuote
                             , autoSelecting: state.props.autoSelecting
                             , searchExpire: state.props.searchExpire
+                            , showProgress : if state.props.currentStage == TryAgain then false else true
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:708]: it can be `not state.props.currentStage == TryAgain`

**Review by witcher-shailesh — CHANGES_REQUESTED** [review]

File: Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs [hunk 4]
```diff
@@ -705,6 +705,7 @@ quoteListModelViewState state = { source: state.data.source
                             , selectedQuote: state.props.selectedQuote
                             , autoSelecting: state.props.autoSelecting
                             , searchExpire: state.props.searchExpire
+                            , showProgress : if ( (DA.null state.data.quoteListModelState) && isLocalStageOn FindingQuotes && state.props.currentStage /= TryAgain) then true else false
```
Comment (witcher-shailesh) [inline Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:708]: We can just have `(DA.null state.data.quoteListModelState) && isLocalStageOn FindingQuotes && state.props.currentStage /= TryAgain` no need to have if else case?

Comment (ashishsingh101) [reply Frontend/ui-customer/src/Screens/RideBookingFlow/HomeScreen/ComponentConfig.purs:708]: yes

**Pull request merged**